src/commands/force/data/soql/query.ts

```diff
@@ -33,19 +36,44 @@ const commonMessages = Messages.loadMessages('@salesforce/plugin-data', 'message
  */
 export class SoqlQuery {
   public async runSoqlQuery(connection: Connection, query: string, logger: Logger): Promise<SoqlQueryResult> {
+    const config: SfdxConfigAggregator = await SfdxConfigAggregator.create();
+
     let columns: Field[] = [];
     logger.debug('running query');
 
-    const result = await connection.query(query, { autoFetch: true, maxFetch: 50000 });
-    logger.debug(`Query complete with ${result.totalSize} records returned`);
-    if (result.totalSize) {
+    // take the limit from the config, then default 10,000
+    const queryOpts: Partial<QueryOptions> = {
+      autoFetch: true,
+      maxFetch: (config.getInfo('maxQueryLimit').value as number) || 10000,
+    };
+
+    const records: Record[] = [];
+
+    const result = await connection.query(query, queryOpts).on('record', (rec) => records.push(rec));
+
+    const totalSize = getNumber(result, 'totalSize', 0);
```

Comment:
It seems to me like there should be some sort of check for the edge case where `result.totalSize` doesn't exist because it we just set it to 0 the message users get will not make any sense.


It'll be something like this:
```
The query result is missing -37892 records due to a 10,000 record limit. Increase the number of records returned by setting the config value "maxQueryLimit" or the environment variable "SFDX_MAX_QUERY_LIMIT" to 0 or greater than 10,000.
```